addons/hr_holidays/static/src/components/thread_icon/thread_icon_tests.js

```diff
@@ -0,0 +1,214 @@
+        const notification = [[false, 'mail.channel', 20], typingData];
+        this.widget.call('bus_service', 'trigger', 'notification', [notification]);
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadIcon_typing',
+        "should have thread icon with partner currently typing"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_ThreadIcon_typing').title,
+        "Demo is typing...",
+        "title of icon should tell demo is currently typing"
+    );
+
+    // simulate receive typing notification from demo "no longer is typing"
+    await afterNextRender(() => {
+        const typingData = {
+            info: 'typing_status',
+            is_typing: false,
+            partner_id: 17,
+            partner_name: "Demo",
+        };
+        const notification = [[false, 'mail.channel', 20], typingData];
+        this.widget.call('bus_service', 'trigger', 'notification', [notification]);
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadIcon_online',
+        "should have thread icon with partner im status icon 'online' (no longer typing)"
+    );
+});
+
+QUnit.test('on leave & online', async function (assert) {
+    assert.expect(2);
+
+    this.data['res.partner'].records.push({
+        id: 7,
+        im_status: 'leave_online',
+        name: 'Demo',
+    });
+     this.data['mail.channel'].records.push({
+        channel_type: 'chat',
+        id: 20,
+        members: [this.data.currentPartnerId, 7],
+    });
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel',
+    });
+    await this.createThreadIcon(thread);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadIcon_online',
+        "thread icon should have online status rendering"
+    );
+    assert.hasClass(
+        document.querySelector('.o_ThreadIcon_online'),
+        'fa-plane',
+        "thread icon should have leave status rendering"
+    );
+});
+
+QUnit.test('on leave & away', async function (assert) {
+    assert.expect(2);
+
+    this.data['res.partner'].records.push({
+        id: 7,
+        im_status: 'leave_away',
+        name: 'Demo',
+    });
+     this.data['mail.channel'].records.push({
+        channel_type: 'chat',
+        id: 20,
+        members: [this.data.currentPartnerId, 7],
+    });
+    await this.start();
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel',
+    });
+    await this.createThreadIcon(thread);
+
+    assert.containsOnce(
+        document.body,
+        '.o_ThreadIcon_away',
+        "thread icon should have away status rendering"
+    );
+    assert.hasClass(
+        document.querySelector('.o_ThreadIcon_away'),
+        'fa-plane',
+        "thread icon should have leave status rendering"
+    );
+});
+
+QUnit.test('on leave & offline', async function (assert) {
```

Comment:
Would precise in test name that this is something about the thread icon of a chat when the correspondent is <im_status>.

At a glance it's redundant with partner im status icon logic and tests. The actual code of thread icon doesn't reuse partner im status icon, which is a shame. I wanted to change that, but it's not that easy hence not already done...